### PR TITLE
[BOOST-4839] feat(sdk): ManagedBudgetWithFees sdk impl

### PR DESCRIPTION
**Pull request by topocount**
This changeset implements the functionality needed to interact with
ManagedBudgetWithFees from the sdk. In the core constructor I've also
added the core address to the authorized list to minimize the chances of
a footgun where BoostCore is not configured correctly in the budget.

### CODE REVIEW

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Your PR body must reference a Github issue using a valid keyword, or your PR title must include the internal Boost ticket number, ie: `[BOOST-1234] Innovate like nuts`
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c553ddb998ac90af2d0b5178976b616764f23e35
</p>

File: notes.md
[the remark this answers is not in the document]
Comment (topocount) [reply notes.md:1]: nope lol

File: packages/sdk/src/Budgets/ManagedBudgetWithFees.test.ts [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+        claimant,
+        boostId: boost.id,
+      });
+    await fixtures.core.claimIncentive(
+      boost.id,
+      0n,
+      trustedSigner.account,
+      claimDataPayload,
+    );
+    let originalBalance = await erc20.balanceOf(accounts[0].account);
```
Comment (jonathandiep) [inline packages/sdk/src/Budgets/ManagedBudgetWithFees.test.ts:238]: Doesn't `ManagedFeeWithBudget` pays out to the boost owner (which is `accounts[4].account`)? Shouldn't we check original and expected balance of the owner?

Comment (topocount) [reply packages/sdk/src/Budgets/ManagedBudgetWithFees.test.ts:238]: Yeah i dug into this and there is no support for setting the owner in the SDK or test harness pipeline yet. It always defaults to `msg.sender`, which i think is desired behavior most of the time, so I figured we'd let it fly. I forgot to remove this after discovering that it's not easily settable in the sdk yet.

**Pull request merged**